aiida/orm/nodes/node.py

```diff
@@ -36,6 +36,46 @@
 _NO_DEFAULT = tuple()
 
 
+class WarnWhenNotEntered:
```

Comment:
Since this should be completely internal, maybe naming it `_WarnWhenNotEntered` would be appropriate?

Reply:
I am not a fan of prefixing resources with an underscore to indicate publicness of the API. We also don't maintain this currently in the code base, so I didn't for consistency. Since I didn't add it to the `__all__` by definition, according to our rules, it is not part of the public API.